compiler/packagehandling.nim

```diff
@@ -7,21 +7,13 @@
 #    distribution, for details about the copyright.
 #
 
-iterator myParentDirs(p: string): string =
-  # XXX os's parentDirs is stupid (multiple yields) and triggers an old bug...
```

Comment:
Your new better parentDirs still has multiple yields and contains logic I don't need.

Reply:
as discussed here https://irclogs.nim-lang.org/04-02-2019.html#10:41:49 let's ensure correctness first and defer to followup PRS to address that point if we find out it's a performance/code size bottlenecks; there are options (at cost of a bit uglier code)

Reply:
The point is: Don't make the compiler's code worse, ``myParentDirs`` still is better than the one in os.nim. Undo the change to compiler/packagehandling.nim

Reply:
done, reverted; we can fix that 'multiple yield' in subsequent PRs, i can think of 2 ways